### PR TITLE
Follow nanocurrency/nano-node#1981

### PR DESCRIPTION
**Pull request by guilhermelawless**
https://github.com/nanocurrency/nano-node/pull/1981 and previous PRs.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -272,11 +277,13 @@ impl RpcService {
                         println!("work_generate completed in {}ms for difficulty {:#x}",
                             start.to(end).num_milliseconds(),
                             difficulty);
-                        let work: Vec<u8> = work.iter().rev().cloned().collect();
```
Comment (guilhermelawless) [inline src/main.rs]: @PlasmaPower it seems to work without this line, what do you think?

**Comment by guilhermelawless** [comment]
e537acf is to avoid panicking during operation, since `println` does not have any error handling. I was using it under a python subprocess with a pipe (other pipes would apply as well) and saw it break multiple times, crashing the server.

File: src/main.rs (hunk 1 above)
Comment (PlasmaPower) [reply src/main.rs]: That line reverses the order of the bytes. Are you sure the work it outputs is still valid without that?

File: src/main.rs [hunk 2]
```diff
@@ -269,14 +275,17 @@ impl RpcService {
                 Box::new(self.generate_work(root, difficulty).then(move |res| match res {
                     Ok(work) => {
                         let end = PreciseTime::now();
-                        println!("work_generate completed in {}ms for difficulty {:#x}",
+                        writeln!(io::stdout(), "work_generate completed in {}ms for difficulty {:#x}",
                             start.to(end).num_milliseconds(),
-                            difficulty);
-                        let work: Vec<u8> = work.iter().rev().cloned().collect();
+                            difficulty)
+                            .unwrap_or(());
```
Comment (PlasmaPower) [inline src/main.rs]: Idiomatic Rust is usually `let _ = writeln!(...);` to throw away errors. I'd use that instead.

File: src/main.rs [hunk 3]
```diff
@@ -295,18 +304,19 @@ impl RpcService {
...
             RpcCommand::WorkCancel(root) => {
-                println!("Received work_cancel");
+                writeln!(io::stdout(), "Received work_cancel").unwrap_or(());
                 self.cancel_work(root);
                 Box::new(Box::new(future::ok((StatusCode::Ok, json!({})))))
             }
             RpcCommand::WorkValidate(root, work, difficulty) => {
-                println!("Received work_validate");
-                let (valid, value) = work_valid(root, work, difficulty);
+                writeln!(io::stdout(), "Received work_validate").unwrap_or(());
```
Comment (PlasmaPower) [inline src/main.rs]: Same as above, I'd prefer `let _ = writeln!(...);`

File: src/main.rs (hunk 2 above)
Comment (guilhermelawless) [reply src/main.rs]: Thank you! Interestingly that workaround is not reported in https://github.com/rust-lang/rust/issues/24821

Comment (guilhermelawless) [reply src/main.rs]: Correct about the vector, reverted the change/

**Comment by guilhermelawless** [comment]
`multiplier` is not accepted in the place of `difficulty`. Might be a possible extension to make it easier to use. Thank you for the review.

File: src/main.rs (hunk 2 above)
Comment (PlasmaPower) [reply src/main.rs]: For the vector I think it should actually use https://doc.rust-lang.org/std/vec/struct.Vec.html#method.reverse instead, which is a lot clearer about what it does.

**Pull request merged**